jvm-packages/project/Dependencies.scala

```diff
@@ -0,0 +1,44 @@
+import sbt._
+
+object Dependencies {
+  val commonDependencies: Seq[ModuleID] = Seq(
+    "commons-logging" % "commons-logging" % "1.2",
+    "org.scalatest" %% "scalatest" % "3.0.0" % "test"
+  )
+
+  val flakkaVersion = "2.3-custom"
+
+  val coreDependencies: Seq[ModuleID] = commonDependencies ++ Seq(
+    // TODO: remove Kryo dependency.
+    "com.esotericsoftware" % "kryo-shaded" % "3.0.3",
+    "com.data-artisans" %% "flakka-actor" % flakkaVersion,
+    "com.data-artisans" %% "flakka-testkit" % flakkaVersion % "test",
+    "junit" % "junit" % "4.11" % "test"
+  )
+
+  val sparkVersion = "2.1.0"
+
+  val sparkDependencies: Seq[ModuleID] = commonDependencies ++ Seq(
+    ("org.apache.spark" %% "spark-core" % sparkVersion).
```

Comment:
can you make spark packages provided?
it will simplify using a different spark version

or even better, use the sbt spark plugin:

Reply:
Good idea, will do!

Reply:
@lev112 could you explain how sbt-spark-package would simplify using different Spark versions? I thought it could cross-package the project against multiple versions, but this feature is not mentioned in the README.

Reply:
my 2 cents here: can we try to avoid relying on a company-owned package? and if we decide to allow user-specified spark versions, let's introduce tests against different versions like https://github.com/graphframes/graphframes/blob/master/.travis.yml#L11....(this could be done in a separate PR)

Reply:
@superbobry setting the spark dependencies as provided, will cause the users spark version to be picked up, I don't know of a simple way to cross package against multiple versions of spark.

even the databricks packages (like https://github.com/databricks/spark-avro#requirements) only support the latest spark version in each realse

setting the dependencies as provided has the added value, that if in the users deployment, the spark dependency jars are on the cluster, they won't be included in the application assembly

I suggested the sbt spark plugin as it can reduce some of the boilerplate code of writing spark package, but AFAIK it doesn't add some significant features

@CodingCat this package is compiled against spark 2.1.0 . If I use 2.0.0, I will just manually exclude all of the spark dependencies, and try it out.
I think it's OK to have 1 tested version (although it should be the latest - 2.2.0), but make it easy for the user to try different versions at his own risk

Reply:
Thanks for the explanation. I think for now it's indeed best to go with provided + cross-tests against Spark versions on Travis (however, I doubt we would have any version-specific regressions).

Reply:
@CodingCat marking Spark as provided would effectively remove it from -with-dependencies JARs. Is this OK for you?

Reply:
yes, I think it's good to make the generated jar thinner